src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultEditorTemplates.cs

```diff
@@ -374,6 +374,22 @@ public static string NumberInputTemplate(IHtmlHelper htmlHelper)
             return GenerateTextBox(htmlHelper, inputType: "number");
         }
 
+        public static string FileInputTemplate(IHtmlHelper htmlHelper)
+        {
+            return GenerateTextBox(htmlHelper, inputType: "file");
+        }
+
+        public static string FileCollectionInputTemplate(IHtmlHelper htmlHelper)
+        {
+            var htmlAttributes = new Dictionary<string, object>()
+            {
+                { "multiple", "" }
+            };
+            htmlHelper.ViewData[HtmlAttributeKey] = htmlAttributes;
+
+            return FileInputTemplate(htmlHelper);
```

Comment:
need to use a different helper / approach, not `GenerateTextBox`.  as-is you're overwriting whatever was passed in the `@Html.Editor(... additionalViewData: X)` argument and using that to get additional information down to `GenerateTextBox`.

suggest instead
``` c#
  var htmlAttributes =
                CreateHtmlAttributes(htmlHelper, className: "text-box single-line", inputType: "file");
  htmlAttributes["multiple"] = "multiple";
  return htmlHelper.TextBox(
                current: null,
                value: value,
                format: null,
                htmlAttributes: htmlAttributes)
                    .ToString();
```